### PR TITLE
Node pool name customisation

### PR DESCRIPTION
**Pull request by melzayet**
## PR Summary

User node pool name should be customisable 

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] This PR is ready to merge and is not **Work in Progress**
- [x] Link to a filed issue: 
- [x] Screenshot of UI changes (if PR includes UI changes)

As described in issue https://github.com/Azure/AKS-Construction/issues/461, user node pool is made configurable

image

In case of only system node pools, the parameter is not set. 

If no value provided by user, it defaults to npuser01, the original string that was used

### CODE REVIEW

**Comment by Gordonby** [comment]
Logic looks good.
I'll fire up a couple of tests then we can merge.  Great PR 😀

**Comment by melzayet** [comment]
@Gordonby one thing I am was not sure of is whether the text input field should be required (with the red asterik).

It's required unless only system node pool exists. It's currently always required. What's your thoughts?

**Comment by Gordonby** [comment]
I actually think we can lean into the system pool a little more.

In the scenario of just a system pool being deployed;
- We relabel the textbox to "Node Pool Name" 
- We allow the system pool to also be renamed
image

Thoughts?

**Review by Gordonby — CHANGES_REQUESTED** [review]
Can we add an output of the pool name(s) ?
This way if a consuming bicep/pipeline uses AKSC then it'll be easy to grab the poolnames.

**Comment by melzayet** [comment]
@Gordonby suggestions absolutely make sense. I have implemented, tested and committed those.

I just want to highlight the commenting out of code block startling line 50 in clusterTab.js. This was clearing vmSKU without any changes in compute type

File: helper/src/components/deployTab.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default function DeployTab({ defaults, updateFn, tabValues, invalidArray,
     ...(cluster.AksPaidSkuForSLA !== defaults.cluster.AksPaidSkuForSLA && { AksPaidSkuForSLA: cluster.AksPaidSkuForSLA }),
     ...(cluster.SystemPoolType === 'none' ? { JustUseSystemPool: true } : cluster.SystemPoolType !== defaults.cluster.SystemPoolType && { SystemPoolType: cluster.SystemPoolType }),
     ...(cluster.vmSize !== defaults.cluster.vmSize && { agentVMSize: cluster.vmSize }),
+    ...(((cluster.nodepoolName.toLowerCase() !== defaults.cluster.nodepoolName  && cluster.SystemPoolType !== 'none') || ( cluster.SystemPoolType === 'none' && cluster.nodepoolName.toLowerCase() !== defaults.cluster.systemNodepoolName )) && { nodePoolName: cluster.nodepoolName }),
```
Comment (Gordonby) [inline helper/src/components/deployTab.js:35]: I think this logic could be simplified now that the name can be used for the system node.

Current logic leads to including the nodepool param when justsystempool is selected.

image

Maybe this works?

```javascript
...(cluster.nodepoolName.toLowerCase() !== defaults.cluster.nodepoolName && { nodePoolName: cluster.nodepoolName })
```

Comment (melzayet) [reply helper/src/components/deployTab.js]: This is now handled indeed in same line you referred to. User doesn't have to provide any value if JustUseSysmPool and did not supply a custom name

**Review by Gordonby — DISMISSED** [review]

**Review by Gordonby — APPROVED** [review]

**Pull request merged**